tools/build_defs/pkg/make_rpm.py

```diff
@@ -155,7 +157,16 @@ class NoRpmbuildFound(Exception):
   pass
 
 
+class InvalidRpmbuild(Exception):
+  pass
+
+
 def FindRpmbuild():
+  if FLAGS.rpmbuild:
+    if not IsExe(FLAGS.rpmbuild):
+      raise InvalidRpmbuild('{} is not executable'.format(FLAGS.rpmbuild))
+  else:
```

Comment:
This isn't right. It is returning FLAGS.rpmbuild if it is empty.

I'm not even sure how it is working for you, since FLAGS turns out not to be globally defined. 

Can you change this to match the existing style.
- pass FLAGS.rpmbuild to the RpmBuilder instantiation in main
- pass rpmbuild to this method so we don't reference FLAGS here.

Reply:
@aiuto addressed in https://github.com/bazelbuild/bazel/pull/7039/commits/19412dad56db75266137d46c374b513cd7af6732

>This isn't right. It is returning FLAGS.rpmbuild if it is empty.

Messed up indentation. `FLAGS.rpmbuild` should only be returned if it's both non-empty AND executable.

>I'm not even sure how it is working for you, since FLAGS turns out not to be globally defined.

Isn't it set globally on [this](https://github.com/bazelbuild/bazel/blob/master/tools/build_defs/pkg/make_rpm.py#L295) line?

>Can you change this to match the existing style.
>pass FLAGS.rpmbuild to the RpmBuilder instantiation in main
>pass rpmbuild to this method so we don't reference FLAGS here.

Done

Reply:
FLAGS gets set globally only if you run make_rpm.py as a main. When used from the tests it is not defined.